libraries/botbuilder-ai/src/luisRecognizer.ts

```diff
@@ -417,40 +428,83 @@ export class LuisRecognizer implements LuisRecognizerTelemetryClient {
         telemetryProperties?: Record<string, string>,
         telemetryMetrics?: Record<string, number>,
         options?: LuisRecognizerOptionsV2 | LuisRecognizerOptionsV3 | LuisPredictionOptions
+    ): Promise<RecognizerResult>;
+
+    /**
+     * Calls LUIS to recognize intents and entities in a users utterance.
+     *
+     * @param {string} utterance The utterance to be recognized.
+     * @param {LuisRecognizerOptionsV2 | LuisRecognizerOptionsV3 | LuisPredictionOptions} options (Optional) options object used to override control predictions. Should conform to the [LuisRecognizerOptionsV2] or [LuisRecognizerOptionsV3] definition.
+     */
+    public async recognize(
+        utterance: string,
+        options?: LuisRecognizerOptionsV2 | LuisRecognizerOptionsV3 | LuisPredictionOptions
+    ): Promise<RecognizerResult>;
+
+    /**
+     * @internal
+     */
+    public async recognize(
+        contextOrUtterance: DialogContext | TurnContext | string,
+        maybeTelemetryPropertiesOrOptions?:
+            | Record<string, string>
+            | LuisRecognizerOptionsV2
+            | LuisRecognizerOptionsV3
+            | LuisPredictionOptions,
+        maybeTelemetryMetrics?: Record<string, number>,
+        maybeOptions?: LuisRecognizerOptionsV2 | LuisRecognizerOptionsV3 | LuisPredictionOptions
     ): Promise<RecognizerResult> {
-        const turnContext = context instanceof DialogContext ? context.context : context;
-        const cached = turnContext.turnState.get(this.cacheKey);
-        const luisRecognizer = options ? this.buildRecognizer(options) : this.luisRecognizerInternal;
-        if (!cached) {
-            const utterance = turnContext.activity.text || '';
-            let recognizerPromise: Promise<RecognizerResult>;
-
-            if (!utterance.trim()) {
-                // Bypass LUIS if the activity's text is null or whitespace
-                recognizerPromise = Promise.resolve({
-                    text: utterance,
-                    intents: { '': { score: 1 } },
-                    entities: {},
-                });
-            } else {
-                recognizerPromise = luisRecognizer.recognizeInternal(context);
-            }
+        if (typeof contextOrUtterance === 'string') {
```

Comment:
I would add another comment here explaining that this type check logically implies that the function is being invoked as the two-argument string + optional options variant.